core/src/main/java/smile/neighbor/CoverTree.java

```diff
@@ -202,6 +202,27 @@ E getObject() {
         public int compareTo(DistanceNode o) {
             return (int) Math.signum(dist - o.dist);
         }
+
+        @Override
+        public boolean equals(Object obj) {
+            if (obj == this) {
+                return true;
+            }
+            if (obj == null) {
+                return true;
+            }
+            if (this.getClass() == obj.getClass()) {
```

Comment:
Actually that rises a compile time exception "Illegal generic type for instanceof", after I searched the problem, this was one of the solutions, the other solution is to make the class "DistanceNode" static. Should I make it static?